src/Stratis.Bitcoin.Features.BlockStore/BlockStoreSignaled.cs

```diff
@@ -124,19 +149,39 @@ protected override void OnNextCore(Block block)
         /// </para>
         /// TODO: consider moving the relay logic to the <see cref="LoopSteps.ProcessPendingStorageStep"/>.
         /// </remarks>
-        public void RelayWorker()
+        private async Task RelayWorkerAsync()
         {
             this.logger.LogTrace("()");
 
-            this.asyncLoop = this.asyncLoopFactory.Run($"{this.name}.RelayWorker", async token =>
+            while (!this.nodeLifetime.ApplicationStopping.IsCancellationRequested)
             {
                 this.logger.LogTrace("()");
 
+                try
+                {
+                    // Wait until a new block is added to the queue.
+                    this.blockEnqueued.Wait(this.nodeLifetime.ApplicationStopping);
+
+                    // Make sure that at least 'FlushFrequencyMilliseconds' passed since the last announcement.
+                    // This is needed in order to ensure announcing blocks in batches to reduce the overhead.
+                    int msSinceLastAnnounce = this.lastBlockAnnounceTimeStamp == DateTime.MinValue ? int.MaxValue :
+                        (int)(this.dateTimeProvider.GetUtcNow() - this.lastBlockAnnounceTimeStamp).TotalMilliseconds;
+                    if (msSinceLastAnnounce < FlushFrequencyMilliseconds)
+                        await Task.Delay(FlushFrequencyMilliseconds - msSinceLastAnnounce, this.nodeLifetime.ApplicationStopping);
```

Comment:
every await we have in FN code should be with .ConfigureAwait(false)

Reply:
Fixed (except for await inside the async loop, probably don't need it here)

Reply:
gentlemen's agreement was to use it everywhere in FN code
but that async loop i don't like